### PR TITLE
Feature/equipment list

### PR DESCRIPTION
**Pull request by ritualz**
There might be some minor bugs and some view inconsistency.

### CODE REVIEW

File: app/views/dashboard/index.html.erb [hunk 1]
```diff
@@ -20,7 +20,7 @@
     </a>
   </div>
   <div class="col-sm-offset-1 col-sm-2">
-    <a href="<%=new_trip_path%>" class="widget">
+    <a href="<%=new_trip_path%>" class="widget", id="create_trip">
```
Comment (fsauter) [inline app/views/dashboard/index.html.erb]: No comma in html :)

File: app/views/equipment_assignments/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<div class='block'>
+    <div class='block-title'>
+        <strong><%= t('Assign item') %></strong>
```
Comment (fsauter) [inline app/views/equipment_assignments/_form.html.erb]: trip_equipment_assignments_assign

File: app/views/equipment_assignments/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            if the current_user is authorized, he/she can add assigments for other people
+            It's done by adding the email for the participant into the field, if the field is blank it's assigned to current_user
+        -->
+        <% if policy(@trip).update? %>
+            <%= text_field_tag(:user_email, '', class: 'form-control') %>
+        <% end %>
+        <!-- Begin slider -->
+        <div class="input-slider-info">
+            <div class="slider slider-horizontal">
+                <div class="slider-track">
```
Comment (fsauter) [inline app/views/equipment_assignments/_form.html.erb]: I think the style="..%.." things comes due to the copy from the console. Can you check if it's the same when you check the source code window? You remember the same as with the dropdown.

File: app/views/equipment_assignments/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<% content_for :page_title do %>
+    equipment | <%= @equipment_list.name %> | <%= @equipment_item.name %>
```
Comment (fsauter) [inline app/views/equipment_assignments/edit.html.erb]: Translation

File: app/views/equipment_assignments/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                It's done by adding the email for the participant into the field, if the field is blank it's assigned to current_user
+            -->
+            <% if policy(@trip).update? %>
+                <%= text_field_tag(:user_email, User.find(@equipment_assignment.user_id).email, class: 'form-control') %>
+            <% end %>
+            <!-- Begin slider -->
+            <div class="input-slider-info">
+                <div class="slider slider-horizontal">
+                    <div class="slider-track">
+                        <div style="left: 0%; width: 42%;" class="slider-selection"></div>
```
Comment (fsauter) [inline app/views/equipment_assignments/edit.html.erb]: same with the style things here

File: app/views/equipment_items/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<div class="row">
+<div class="col-md-10 col-md-offset-1 col-lg-8 col-lg-offset-2">
+    <div class="block">
+        <!-- Horizontal form title -->
+        <div class="block-title">
+            <h2>
+                <% if current_page?(:action => 'new') %>
+                    Create equipment item for <%= @equipment_list.name %>
```
Comment (fsauter) [inline app/views/equipment_items/_form.html.erb]: Translation

File: app/views/equipment_items/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            <% else %>
+                Edit <%= @equipment_item %>
```
Comment (fsauter) [inline app/views/equipment_items/_form.html.erb]: Translation

File: app/views/equipment_items/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            </h2>
+        </div>
+        <!-- END horizontal form title -->
+
+        <%= form_for [@trip, @equipment_list, @equipment_item], html: {class: 'form-horizontal form-bordered'} do |f| %>
+            <div class="form-group">
+                <%= f.label :name, class: 'col-md-3 control-label' %>
+                <div class="col-md-9">
+                    <%= f.text_field :name, class: 'form-control'%>
+                    <span class="help-block">Any equipment item name that's fine.</span>
```
Comment (fsauter) [inline app/views/equipment_items/_form.html.erb]: Translation

File: app/views/equipment_items/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+                <div class="col-md-9">
+                    <%= f.text_area :price, class: 'form-control', placeholder: 'Price of a item.', rows: 7 %>
+                </div>
+            </div>
+
+            <div class="form-group">
+                <%= f.label :number, class: 'col-md-3 control-label' %>
+                <div class="col-md-4">
+                    <%= f.text_field :number, class: 'form-control'%>
+                    <span class="help-block">Number of items</span>
```
Comment (fsauter) [inline app/views/equipment_items/_form.html.erb]: Translation

File: app/views/equipment_items/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                <div class="col-md-9">
+                    <%= f.text_field :name, class: 'form-control'%>
+                    <span class="help-block">Any equipment item name that's fine.</span>
+                </div>
+            </div>
+
+            <div class="form-group">
+                <%= f.label :price, class: 'col-md-3 control-label' %>
+                <div class="col-md-9">
+                    <%= f.text_area :price, class: 'form-control', placeholder: 'Price of a item.', rows: 7 %>
```
Comment (fsauter) [inline app/views/equipment_items/_form.html.erb]: Translation

File: app/views/equipment_items/_form_small.html.erb [hunk 11]
```diff
@@ -0,0 +1,29 @@
+
+
+<%= form_for [@trip, @equipment_list, @equipment_item], html: {class: 'form-horizontal form-bordered'} do |f| %>
+    <div class="form-group">
+        <div class="col-md-8">
+            <%= f.text_field :name, class: 'form-control input-lg', placeholder: 'Name of new item' %>
```
Comment (fsauter) [inline app/views/equipment_items/_form_small.html.erb]: Translation

File: app/views/equipment_items/_form_small.html.erb [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        </div>
+    </div>
+
+    <div class="form-group">
+        <div class="col-md-8">
+            <%= f.number_field :price, class: 'form-control input-lg', placeholder: 'Price of a item.' %>
```
Comment (fsauter) [inline app/views/equipment_items/_form_small.html.erb]: Translation

File: app/views/equipment_items/_form_small.html.erb [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+        </div>
+    </div>
+
+    <div class="form-group">
+        <div class="col-md-8">
+            <%= f.number_field :number, class: 'form-control input-lg', placeholder: 'Number of items' %>
```
Comment (fsauter) [inline app/views/equipment_items/_form_small.html.erb]: Translation

File: app/views/equipment_items/edit.html.erb [hunk 14]
```diff
@@ -0,0 +1,10 @@
+<% content_for :page_title do %>
+    | <%= @equipment_list.name %> | new item
```
Comment (fsauter) [inline app/views/equipment_items/edit.html.erb]: Translation

File: app/views/equipment_items/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<% content_for :page_title do %>
+    | <%= @equipment_list.name %> | new item
```
Comment (fsauter) [inline app/views/equipment_items/new.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,116 @@
+<% content_for :page_title do %>
+    equipment | <%= @equipment_list.name %> | <%= @equipment_item.name %>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+
+
+<%= render 'equipment_lists/sidebar' %>
+
+<% content_for :content_header do %>
+    <div class="content-header">
+      <div class="row">
+        <div class="col-sm-12">
+          <div class="header-section">
+              <h1><%= @equipment_item.name %></h1><i>Created by: <%= User.find(@equipment_item.user_id).name %></i>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+          </div>
+
+        </div>
+      </div>
+    </div>
+<% end %>
+
+<!-- EquipmentList -->
+<div class='block' id='EquipmentList'>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: ids like this: equipment-list-container or just equipment-list

File: app/views/equipment_items/show.html.erb [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+        <div class='block-options pull-right'>
+            <div class='btn-group'>
+                <% if policy(@trip).update? %>
+                    <a href="javascript:void(0)" class="btn btn-effect-ripple btn-primary dropdown-toggle enable-tooltip" data-toggle="dropdown" title="Extras"><i class="fa fa-chevron-down"></i></a>
+
+                    <ul class="dropdown-menu dropdown-menu-right">
+                        <li>
+                            <a href="<%= edit_trip_equipment_list_equipment_item_path(@trip, @equipment_list, @equipment_item) %>", id='edit_equipment_item'>
+                                <i class="fa fa-pencil pull-right"></i>
+                                edit
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+                            </a>
+                        </li>
+                        <% if current_user.id == @equipment_item.user_id or policy(@trip).destroy? %>
+                            <li class="divider"></li>
+                            <li>
+                                <%= link_to raw('<i class="fa fa-trash-o fa-fw pull-right"></i> delete'),
+                                    trip_equipment_list_equipment_item_path(@trip, @equipment_list, @equipment_item),
+                                    :method => :delete,
+                                    :data => {
+                                        :confirm    => 'are you sure you want to delete this equipment item?',
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+                                        :toogle     => 'tooltip'
+                                    },
+                                    :class => 'text-danger',
+                                    :title => '',
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Title

File: app/views/equipment_items/show.html.erb [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+                                %>
+                        </li>
+                    <% end %>
+                    </ul>
+                <% end %>
+            </div>
+            <!-- end drop down menu -->
+        </div>
+    </div>
+    <p> <h3> Number of items </h3> <%= @equipment_item.number %> </p>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+    <p> <h3> Price </h3> <%= @equipment_item.price %> </p>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+
+
+<!-- begin equipment table if there is any assigments -->
+<% if @equipment_item.equipment_assignments.count != 0 %>
+    <div class='block'>
+        <div class='table-responsive'>
+            <table class="table table-borderless table-vcenter">
+                <thead>
+                    <tr>
+                        <th>Assigned to</th>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+                            <th>Number</th>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+                                    <td class="text-center">
+                                        <% if policy(@trip).update? %>
+                                            <a href="<%= edit_trip_equipment_list_equipment_item_equipment_assignment_path(@trip, @equipment_list, @equipment_item, assignment) %>" data-toggle="tooltip" id='edit_equipment_assignment' title="Edit assignment" class="btn btn-effect-ripple btn-xs btn-success"><i class="fa fa-pencil"></i></a>
+                                        <% end %>
+                                        <% if policy(@trip).destroy? %>
+                                            <%= link_to raw('<i data-toggle="tooltip" title="delete item" class="btn btn-effect-ripple btn-xs btn-danger fa fa-times"></i>'),
+                                                trip_equipment_list_equipment_item_equipment_assignment_path(@trip, @equipment_list, @equipment_item, assignment),
+                                                :method => :delete,
+                                                :data => {
+                                                    :confirm => 'Are you sure you want to delete this assignment?',
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Translation

File: app/views/equipment_items/show.html.erb [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+                                                    :toogle => 'tooltip'
+                                                },
+                                                :class => 'text-danger',
+                                                :title => '',
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Title

File: app/views/equipment_items/show.html.erb [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+                    <% end %>
+                </tbody>
+            </table>
+        </div>
+        <!-- end block -->        
+    </div>
+<% end %>
+
+<div class='block'>
+    <% unless @equipment_item.equipment_assignments.sum(:number) == @equipment_item.number %>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: use eql? instead of == maybe?

File: app/views/equipment_lists/_form.html.erb [hunk 29]
```diff
@@ -1,3 +1,19 @@
+<% content_for :page_stylesheets do %>
+    <link rel="stylesheet" href="<%= asset_path 'app/js/vendor/fontawesome-iconpicker-1.0.0/dist/css/fontawesome-iconpicker.min.css' %>">
+<% end %>
+
+<% content_for :page_scripts do %>
+    <script src="<%= asset_path 'app/js/vendor/fontawesome-iconpicker-1.0.0/dist/js/fontawesome-iconpicker.min.js' %>"></script>
+    <script>
+        $(function() {
+            $('#equipment_list_icon').iconpicker({
```
Comment (fsauter) [inline app/views/equipment_lists/_form.html.erb]: We should create a ticket to check why it's not working for firefox and co.

File: app/views/equipment_lists/_form.html.erb [hunk 30]
```diff
@@ -14,18 +30,6 @@
       <!-- END Horizontal Form Title -->
 
       <%= form_for [@trip, @equipment_list], html: {class: 'form-horizontal form-bordered'} do |f| %>
-          <% if @equipment_list.errors.any? %>
-              <div id="error_explanation">
-                <h2> <%= pluralize(@equipment_list.errors.count, "error") %>
-                  prohibited this equipment list from begin saved:
```
Comment (fsauter) [inline app/views/equipment_lists/_form.html.erb]: Translation

File: app/views/equipment_lists/_form.html.erb [hunk 31]
```diff
@@ -38,16 +42,15 @@
           <div class="form-group">
             <%= f.label :description, class: 'col-md-3 control-label' %>
             <div class="col-md-9">
-              <div class="input-group">
-                <%= f.text_field :description, class: 'form-control', placeholder: 'Equipment list description.' %>
-              </div>
+              <%= f.text_area :description, class: 'form-control', placeholder: 'Equipment list description.', rows: 7 %>
```
Comment (fsauter) [inline app/views/equipment_lists/_form.html.erb]: Translation

File: app/views/equipment_lists/_form.html.erb [hunk 32]
```diff
@@ -38,16 +42,15 @@
...
             </div>
           </div>
 
           <div class="form-group">
             <%= f.label :icon, class: 'col-md-3 control-label' %>
-            <div class="col-md-9">
-              <%= f.text_field :icon, class: 'form-control', placeholder: 'Awesome icon for your equipment list...'%>
+            <div class="col-md-4">
+              <%= f.text_field :icon, class: 'form-control'%>
+              <span class="help-block">Awesome icon for your equipment list...</span>
```
Comment (fsauter) [inline app/views/equipment_lists/_form.html.erb]: Translation

File: app/views/equipment_lists/index.html.erb [hunk 33]
```diff
@@ -1,22 +1,31 @@
...
-                            <i class="fa fa-<%= equipment_list.icon %> fa-fw icon-push"></i><strong><%= equipment_list.name %></strong>
-                            </a>
-                        </li>
-                    <% end %>
-                </ul>
-            </div>
-      <!-- end lists -->
-      </div>
+<% content_for :page_title do %>
+    Equipment Lists
```
Comment (fsauter) [inline app/views/equipment_lists/index.html.erb]: Translation

File: app/views/equipment_lists/index.html.erb [hunk 34]
```diff
@@ -1,22 +1,31 @@
...
+<%= render 'sidebar' %>
+
+<div class="row">
+
+  <div class="col-md-offset-3 col-md-6 text-center">
+
+    <div class="clearfix">
+      <blockquote class="pull-left">
+        <% if @trip.equipment_lists.empty? %>
+            <p>You don't have any equipment lists right now.</p>
```
Comment (fsauter) [inline app/views/equipment_lists/index.html.erb]: Translation

File: app/views/equipment_lists/index.html.erb [hunk 35]
```diff
@@ -1,22 +1,31 @@
...
+            <p>
+                <a class="btn btn-danger" href="<%= new_trip_equipment_list_path(@trip) %>">Create a new equipment list</a>
+            </p>
+        <% else %>
+            <p>Click on a equipment list to the left to collaborate.</p>
```
Comment (fsauter) [inline app/views/equipment_lists/index.html.erb]: Translation

File: app/views/equipment_lists/index.html.erb [hunk 36]
```diff
@@ -1,22 +1,31 @@
...
+        <% end %>
+        <small>Yours sincerely, turi team</small>
```
Comment (fsauter) [inline app/views/equipment_lists/index.html.erb]: Translation (exists already)

File: app/views/equipment_lists/show.html.erb [hunk 37]
```diff
@@ -1,49 +1,138 @@
...
+
+<%= render 'sidebar' %>
+
+
+<% content_for :content_header do %>
+    <div class="content-header">
+      <div class="row">
+        <div class="col-sm-12">
+          <div class="header-section">
+              <h1><%= @equipment_list.name %></h1><i>Created by: ...</i>
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: Translation

File: app/views/equipment_lists/show.html.erb [hunk 38]
```diff
@@ -1,49 +1,138 @@
...
+                            </a>
+                        </li>
+                        <% if policy(@trip).destroy? %>
+                            <li class="divider"></li>
+                            <li>
+                                <%= link_to raw('<i class="fa fa-trash-o fa-fw pull-right"></i> Delete'),
+                                    trip_equipment_list_path(@trip, @equipment_list),
+                                    :method => :delete,
+                                    :data => {
+                                        :confirm => 'Are you sure you want to delete this equipment list?',
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: Translation

File: app/views/equipment_lists/show.html.erb [hunk 39]
```diff
@@ -1,49 +1,138 @@
...
+                                    :title => '',
+                                    :id => 'delete_equipment_list'
+                                %>
+                        </li>
+                    <% end %>
+                    </ul>
+                <% end %>
+            </div>
+        </div>
+        <h2> <%= t('Description') %> </h2>
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: t('description')

File: app/views/equipment_lists/show.html.erb [hunk 40]
```diff
@@ -1,49 +1,138 @@
...
+                            <li>
+                                <%= link_to raw('<i class="fa fa-trash-o fa-fw pull-right"></i> Delete'),
+                                    trip_equipment_list_path(@trip, @equipment_list),
+                                    :method => :delete,
+                                    :data => {
+                                        :confirm => 'Are you sure you want to delete this equipment list?',
+                                        :toogle => 'tooltip'
+                                    },
+                                    :class => 'text-danger',
+                                    :title => '',
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: Title

File: app/views/equipment_lists/show.html.erb [hunk 41]
```diff
@@ -1,49 +1,138 @@
...
+                        <td class="text-center">
+                            <% if policy(@trip).update? %>
+                                <a href="<%= edit_trip_equipment_list_equipment_item_path(@trip, @equipment_list, item) %>" data-toggle="tooltip" title="Edit item" class="btn btn-effect-ripple btn-xs btn-success"><i class="fa fa-pencil"></i></a>
+                            <% end %>
+                            <% if policy(@trip).destroy? %>
+                                <%= link_to raw('<i data-toggle="tooltip" title="delete item" class="btn btn-effect-ripple btn-xs btn-danger fa fa-times"></i>'),
+                                    trip_equipment_list_equipment_item_path(@trip, @equipment_list, item),
+                                    :method => :delete,
+                                    :data => {
+                                        :confirm => 'Are you sure you want to delete this equipment item?',
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: Translation

File: app/views/equipment_lists/show.html.erb [hunk 42]
```diff
@@ -1,49 +1,138 @@
...
+                                        :toogle => 'tooltip'
+                                    },
+                                    :class => 'text-danger',
+                                    :title => '',
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: Title

File: app/views/equipment_lists/show.html.erb [hunk 43]
```diff
@@ -1,49 +1,138 @@
...
+                        <% end %>
+                        </td>
+                    </tr>
+                <% end %>
+            </tbody>
+        </table>
+    </div>
+</div>
+<% else %>
+    <%= t('Add item') %>
```
Comment (fsauter) [inline app/views/equipment_lists/show.html.erb]: t('trip_equipment_item_add')

File: public/app/css/main.css [hunk 44]
```diff
@@ -2460,10 +2460,6 @@ a.list-group-item.active > .badge {
 }
 
 .dropdown-menu {
-    z-index: 100;
-}
```
Comment (fsauter) [inline public/app/css/main.css]: Shame on me :stuck_out_tongue_winking_eye:

File: app/views/equipment_lists/_form.html.erb (hunk 29 above)
Comment (ritualz) [inline app/views/equipment_lists/_form.html.erb]: Yes, it does not work for me in FF and Chromium.

**Pull request closed without merging**